pkg/nuclide-server/lib/services/FileSystemService.js

```diff
@@ -255,12 +255,22 @@ export function unlink(path: string): Promise<void> {
  */
 export async function readFile(
   path: string,
-  options?: {flag?: string},
+  options?: {encoding?: string, flag?: string},
 ): Promise<Buffer> {
   const stats = await fsPromise.stat(path);
   if (stats.size > READFILE_SIZE_LIMIT) {
     throw new Error(`File is too large (${stats.size} bytes)`);
   }
+  if (options && options.encoding) {
+    const encoding = options.encoding;
+    delete options.encoding;
+    if (encoding !== 'utf8') {
+      const iconv = require('iconv-lite');
```

Comment:
Nuclide uses a transform to lazy-require all imports, so just import this at the top of the file in the usual way.